cmd/bucket-replication.go

```diff
@@ -1053,9 +1058,19 @@ func replicateObjectToTarget(ctx context.Context, ri ReplicateObjectInfo, object
 			ReplicationProxyRequest: "false",
 		}})
 	if cerr == nil {
-		rAction = getReplicationAction(objInfo, oi)
+		rAction = getReplicationAction(objInfo, oi, ri.OpType)
 		rinfo.ReplicationStatus = replication.Completed
 		if rAction == replicateNone {
+			if ri.OpType == replication.ExistingObjectReplicationType &&
+				objInfo.ModTime.Unix() > oi.LastModified.Unix() && objInfo.VersionID == nullVersionID {
+				logger.LogIf(ctx, fmt.Errorf("Unable to replicate %s/%s (null). Newer version exists on target", bucket, object))
+				sendEvent(eventArgs{
+					EventName:  event.ObjectReplicationNotTracked,
+					BucketName: bucket,
+					Object:     objInfo,
+					Host:       "Internal: [Replication]",
+				})
+			}
```

Comment:
Do we not want to return right away here @poornas ?

Reply:
Not necessarily - we still want to update metadata to `COMPLETED` in case replication status of target is not updated yet from some prior replication activity (on account of 503's) before returning